scripts/configure-ironic.sh

```diff
@@ -70,11 +70,48 @@ if [[ -z "${DEPLOY_KERNEL_URL:-}" ]] && [[ -z "${DEPLOY_RAMDISK_URL:-}" ]] && \
     export DEPLOY_RAMDISK_URL="file://${IMAGE_CACHE_PREFIX}.initramfs"
 fi
 
+declare -A detected_arch
```

Comment:
nit: the entire logic is large enough to get split into a helper file

Reply:
also, as an aside, this is exactly the level of bash complexity where I wish the entire thing was written in Python :)